src/compiler/transformers/classFields.ts

```diff
@@ -211,7 +211,7 @@ namespace ts {
             return visited;
         }
 
-        function visitor(node: Node): VisitResult<Node> {
+        function visitor(node: Node): VisitResult<Node | undefined> {

```

Comment:
This is a bummer. Checking the PR out to see if I can work out another way.

Reply:
Wait, there were already `undefined` returns in here? I don’t understand how it compiled before. This isn’t just a `strictFunctionTypes` thing; a return value of `undefined` shouldn’t have been assignable to `VisitResult<Node>` under plain old `strictNullChecks`. Right??

Reply:
I changed the definition of VisitResult so that the visitor functions correctly indicate the undefined-ness of the output and correctly handle arrays of differing values. Before, it was like:

```
type VisitResult<T extends Node> = T | T[] | undefined
```

And now it's:

```
type VisitResult<T extends Node | undefined> = T | readonly Node[];
```

Which does not always include undefined. If I don't do this, then the NodeVisitor/NodesVisitor types can't figure out if the function returns defined or not.

Reply:
In a previous revision I had all of these cases as `VisitResult<Node> | undefined` which is equivalent in practice, but I figured I'd try and stick it into the right place as indicated by the new type parameter.

I'd love to be able to gauge how breaky this is; this PR is just "visitor API break" central.

Reply:
Yeah, though in practice I think that making an API that can handle that would basically turn us into the codebase from #51188, with type so big we chug. Hence, forcing people to provide narrowing functions (which have the side effect of catching bugs, at least).